ipamapi/contract.go

```diff
@@ -42,59 +51,22 @@ type AddressSpace string
 // Common key is a addressspace
 type Config interface {
```

Comment:
Before reviewing the implementation, I would like to get the API right. So, a few questions.

1. What is the purpose of having a `Config` interface  with Add and Delete Subnet ?
2. Why cant the IPAM Driver/Plugin dynamically create a subnet or remove a subnet during the Allocator's `Request` & `Release` calls ?